src/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useContext, useEffect, useReducer } from "react"
+import { useDidMount } from "Utils/Hooks/useDidMount"
+import { hasFilters } from "./Utils/hasFilters"
+import { isDefaultFilter } from "./Utils/isDefaultFilter"
+import { rangeToTuple } from "./Utils/rangeToTuple"
+import { urlFragmentFromState } from "./Utils/urlFragmentFromState"
+
+export interface Filters {
+  acquireable?: boolean
+  at_auction?: boolean
+  color?: string
+  for_sale?: boolean
+  height: string
+  inquireable_only?: boolean
+  keyword?: string
+  major_periods: string[]
+  medium?: string
+  offerable?: boolean
+  page: number
+  partner_id?: string
+  price_range: string
+  sort: string
+  width: string
+}
+
+export const initialFilterState = {
+  height: "*-*",
+  major_periods: [],
+  page: 1,
+  price_range: "*-*",
+  sort: "-decayed_merch",
+  width: "*-*",
+}
+
+export const FilterContext = React.createContext<{
+  filters: Filters
+  hasFilters: boolean
+  isDefaultValue: (name: string) => boolean
+  rangeToTuple: (name: string) => [number, number]
+  resetFilters: () => void
+  setFilter: (name: string, value: any) => void
+  unsetFilter: (name: string) => void
+}>({
+  filters: initialFilterState,
+  hasFilters: false,
+  isDefaultValue: null,
+  rangeToTuple: null,
+  resetFilters: null,
+  setFilter: null,
+  unsetFilter: null,
+})
+
+export const FilterContextProvider: React.FC<
+  Filters & { children: React.ReactNode }
+> = ({
+  children,
+  height = "*-*",
+  major_periods = [],
+  medium = "",
+  page = 1,
+  price_range = "*-*",
+  sort = "-decayed_merch",
+  width = "*-*",
+  ...rest
+}) => {
+  const [state, dispatch] = useReducer(filterReducer, {
+    height,
+    major_periods,
+    medium,
+    page,
+    price_range,
+    sort,
+    width,
+    ...rest,
+  })
+  const isMounted = useDidMount()
+  // const updateOnChange = Object.keys(state).map(prop => state[prop])
+
+  useEffect(() => {
+    if (isMounted) {
+      const queryString = urlFragmentFromState(state)
+      window.history.pushState(
+        {},
+        null,
+        `${window.location.pathname}?${queryString}`
+      )
+    }
+  })
+
+  /**
+   * TODO: use this to trigger resetting of fields when back/fwd buttons are hit
```

Comment:
@pepopowitz - one thing I noticed is that the `TODO -` form doesn't trigger syntax highlighting in our editors; I updated these notes with the form that they expect (`TODO:` -- the ":" is what does it).